### PR TITLE
Xcm-Simulator Docs

### PR DESCRIPTION
**Pull request by Szegoo**
Closes: #3542

### CODE REVIEW

File: xcm/xcm-simulator/src/lib.rs [hunk 1]
```diff
@@ -202,6 +211,10 @@ thread_local! {
 		= RefCell::new(VecDeque::new());
 }
 
+/// Declares a test network. Expects a network struct as an arugument and
+/// implements some functionality for testing and the `ParachainXcmRouter` and
+/// `RelayChainXcmRouter`. The struct needs to contain the relay chain struct
+/// and an indexed list of parachains that are going to be in the network.
 #[macro_export]
 macro_rules! decl_test_network {
```
Comment (Szegoo) [inline xcm/xcm-simulator/src/lib.rs:219]: Should I provide examples in these comments? @shawntabrizi

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
The job name - test-linux-stable
The job logs - https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/1977700

**Comment by Szegoo** [comment]
@shawntabrizi PTAL.

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
ideally these would not be `ignore` but actually compile, but not gonna stress on incremental improvements.

File: xcm/xcm-simulator/src/lib.rs [hunk 2]
```diff
@@ -108,6 +122,21 @@ macro_rules! decl_test_relay_chain {
 	};
 }
 
+/// Expects a parachain struct as an argument and implements`DmpMessageHandlerT`
```
Comment (bkchr) [inline xcm/xcm-simulator/src/lib.rs:125]: This should document on what this does and not directly on what it expects.

File: xcm/xcm-simulator/src/lib.rs [hunk 3]
```diff
@@ -76,6 +77,19 @@ pub fn encode_xcm(message: Xcm<()>, message_kind: MessageKind) -> Vec<u8> {
 	}
 }
 
+/// Expects a relay chain struct as an argument and implements UMP for it. The
```
Comment (bkchr) [inline xcm/xcm-simulator/src/lib.rs:80]: Same.

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot rebase

**Comment by bkchr** [comment]
bot rebase

**Pull request merged**